js/src/components/paid-ads/campaign-assets-form.js

```diff
@@ -158,12 +158,18 @@ export default function CampaignAssetsForm( {
 					: initialAssetGroup;
 				let hasNonEmptyAssets = false;
 
-				Object.keys( emptyAssetGroup ).forEach( ( key ) => {
-					if ( assetGroup && assetGroup[ key ]?.length ) {
-						hasNonEmptyAssets = true;
-					}
-					formContext.setValue( key, nextAssetGroup[ key ] );
-				} );
+				const updatedContextValues = Object.fromEntries(
+					Object.keys( emptyAssetGroup ).map( ( key ) => {
+						if ( assetGroup?.[ key ]?.length ) {
+							hasNonEmptyAssets = true;
+						}
+						return [ key, nextAssetGroup[ key ] ];
+					} )
+				);
+
+				if ( Object.keys( updatedContextValues ).length ) {
+					formContext.setValues( updatedContextValues );
```

Comment:
On machines with sufficient processing speed, the following code blocks may cause a race condition, where the next `delegation` is dispatched before the previous `delegation` has been processed by the `useEffect` callback. When I wrote this handling previously, my machine was too slow to notice this issue.

https://github.com/woocommerce/google-listings-and-ads/blob/508b848a36ea5cf4122bbd9dbd6208535168bb95/js/src/components/adaptive-form/adaptive-form.js#L190-L199

https://github.com/woocommerce/google-listings-and-ads/blob/508b848a36ea5cf4122bbd9dbd6208535168bb95/js/src/components/adaptive-form/adaptive-form.js#L64-L68

Since compatibility with WC < 7.1 is no longer required, it's okay to call `setValues` directly to avoid that race condition.